docs/design/2022-06-15-phyical_plan_explain.md

```diff
@@ -0,0 +1,684 @@
+   ```
+
+   ```text
+   == Physical Plan ==
+   *(1) ColumnarToRow
+   +- TiKV CoprocessorRDD{[table: t3] TableScan, Columns: a@LONG, KeyRange: [([t\200\000\000\000\000\000\005/_r\200\000\000\000\000\000\000\001], [t\200\000\000\000\000\000\005/_s\000\000\000\000\000\000\000\000])], startTs: 434063568626778113}
+   ```
+
+## Detailed Design
+
+### Design Overview
+
+- **`TableRangeScan`**: Table scans with the specified range. We consider full table scan as a special case of `TableRangeScan`, so full table scan is also called `TableRangeScan`.
+
+- **`TableRowIDScan`**: Scans the table data based on the `RowID`. Usually follows an index read operation to retrieve the matching data rows.
+
+- **`IndexRangeScan`**: Index scans with the specified range. We consider full index scan as a special case of `IndexRangeScan`, so full index scan is also called `IndexRangeScan`.
+
+- **`RangeFilter`**: `RangeFilter` indicates which conditions the range is made up of. `RangeFilter` generally appears when the query involves an index. If `RangeFilter` is empty, it indicates a full table scan or full index scan. `RangeFilter` generally appears when the query involves an index range, when query the expressions in the `RangeFilter` form the scanned range from left to right.
+
+  > **Which expression will be appeared in `RangeFilter`**
+  >
+  > 1. iterate through all the columns in the Index from left to right.
+  > 2. If there is an Expression related to the column in Expression and the Expression is dot-checked, add that `Expression` to `RangeFilter`. Go back to step 1.
+  > 3. If there is an Expression related to the column in the Expression and the Expression is a range, add the `Expression` to the `RangeFilter` and exit.
+  > 4. If there is no Expression related to the column in the Expression; exit. Output we expect
+  >
+  > For example:
+  >
+  > ```sql
+  > CREATE TABLE `t1` (
+  >   `a` BIGINT(20)  NOT NULL,
+  >   `b` varchar(255) NOT NULL,
+  >   `c` varchar(255) DEFAULT NULL,
+  >   PRIMARY KEY (`a`,`b`)
+  > ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin
+  > CREATE TABLE `t2` (
+  >   `a` BIGINT(20)  NOT NULL,
+  >   `b` varchar(255) NOT NULL,
+  >   `c` varchar(255) DEFAULT NULL,
+  >   PRIMARY KEY (`b`,`a`)
+  > ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin
+  > ```
+  >
+  > If we execute:
+  >
+  > ```sql
+  > SELECT t1.a FROM t1 WHERE t1.a = 0 and t1.b>'aa'
+  > ```
+  >
+  > Because the primary key of `t1` has `a` in the front and `b` in the back, we can use these two conditions to build a scan range. The `RangeFilter` is `{a=0,b>'aa'}`, the constructed scan range is `(tid_i_a=0_b='aa', tid_i_a_inf)`.
+  >
+  > If we execute:
+  >
+  > ```sql
+  > SELECT t1.a FROM t1 WHERE t1.b>'aa'
+  > ```
+  >
+  > Because the primary key of `t1` has a in the front and b in the back, we can't use `b>'aa'` to build the range of the query.  The `RangeFilter` is empty, The scan range is `(tid_i_-inf,tid_i_+inf)`.
```

Comment:
```suggestion
  > Because the primary key of `t1` has a in the front and b in the back, we can't use `b>'aa'` to build the range of the query. The `RangeFilter` is empty, The scan range is `(tid_i_-inf,tid_i_+inf)`.
```

Reply:
delte